Utils/Utils.hs

```diff
@@ -35,3 +35,10 @@ partition _ [x]      = [[x]]
 partition f (x:y:xs)
   | f x == f y = tack x (partition f (y : xs))
   | otherwise  = [x] : partition f (y : xs)
+
+-- Checks if two (unordered) lists contain the same elements
+-- E.g. listsEqual [3, 2, 1] [2, 1, 3] = True
+-- E.g. listsEqual [5, 5] [4, 4] = False
+areOrPatBindsEqual :: (Ord a, Eq a) => [a] -> [a] -> Bool
+areOrPatBindsEqual xs ys = (xs == nub xs) && (ys == nub ys) &&
+  (sort xs == sort ys)
```

Comment:
We have a function actually that checks for distinctness (I think it's called `Utils.Utils.distinct`). Use that instead of explicitly checking using `nub`.